bin/main.py

```diff
@@ -52,6 +53,15 @@ def run():
         else:
             Thread(target=process, args=(line, None)).start()
 
+def run_ipc():
```

Comment:
This should not be a different code path, you either use the default receive/send functions or you set new ones due to another module being chosen. The main loop should be only one.